src/routes/access.ts

```diff
@@ -10,92 +9,89 @@ const { UnauthorizedError, api } = sbvrUtils;
 
 const HOSTOS_ACCESS_MIN_OS_VER = '2.0.0';
 
-export function hostOSAccess(
+export async function hostOSAccess(
 	req: express.Request,
 	res: express.Response,
 ): Promise<void> {
-	return api.resin
-		.get({
-			resource: 'device',
-			options: {
-				$select: ['id', 'os_version'],
-				$filter: {
-					uuid: req.params['device_uuid'],
-				},
+	const devices = await api.resin.get({
+		resource: 'device',
+		options: {
+			$select: ['id', 'os_version'],
+			$filter: {
+				uuid: req.params['device_uuid'],
 			},
+		},
+		passthrough: {
+			req,
+		},
+	});
+
+	if (!_.isArray(devices)) {
+		res.sendStatus(401);
+		return;
+	}
+
+	if (devices.length !== 1) {
+		res.sendStatus(401);
+		return;
+	}
+
+	const [device] = devices;
+
+	try {
+		const allowedDevices = (await api.resin.post({
+			resource: 'device',
+			id: device.id,
 			passthrough: {
 				req,
 			},
-		})
-		.then(devices => {
-			if (!_.isArray(devices)) {
-				res.sendStatus(401);
-				return;
-			}
-
-			if (devices.length !== 1) {
-				res.sendStatus(401);
-				return;
-			}
-
-			const device = devices[0];
+			body: {
+				action: 'ssh-host',
+			},
+			url: `device(${device.id})/canAccess`,
+		})) as { d?: Array<AnyObject> };
 
-			return api.resin
-				.post({
-					resource: 'device',
-					id: device.id,
-					passthrough: {
-						req,
-					},
-					body: {
-						action: 'ssh-host',
-					},
-					url: `device(${device.id})/canAccess`,
-				})
-				.then((allowedDevices: any) => {
-					if (
-						allowedDevices.d == null ||
-						!_.isArray(allowedDevices.d) ||
-						allowedDevices.d.length !== 1
-					) {
-						res.sendStatus(401);
-						return;
-					}
+		if (
+			allowedDevices.d == null ||
+			!_.isArray(allowedDevices.d) ||
```

Comment:
You can drop the check above, since isArray will take care of it being null-ish.